docs/configuration/single-process-config-blocks-gossip-1.yaml

```diff
@@ -96,3 +96,12 @@ ruler_storage:
   backend: local
   local:
     directory: /tmp/cortex/rules
+
+alertmanager:
+  external_url: http://localhost/alertmanager
+
+alertmanager_storage:
+  backend: local
+  local:
+    # Make sure file exist
```

Comment:
I am wondering if we can improve this a bit. Do we require the folder for rules to exist? Or Ruler will try to create it beforehand

Reply:
Ruler checks the rule file as well, but it doesn't abort. However, the alertmanager abort. So, I added the annotations only to the alertmanager.

The alertmanager logs are as follows:
```
ts=2024-10-18T01:05:01.787107Z caller=cortex.go:450 level=error msg="module failed" module=alertmanager err="invalid service state: Failed, expected: Running, failure: failed to list users with alertmanager configuration: unable to walk file path at /tmp/cortex/alerts: lstat /tmp/cortex/alerts: no such file or directory"

level=error msg="error running cortex" err="failed services\ngithub.com/cortexproject/cortex/pkg/cortex.(*Cortex)
```
The ruler logs:
```
ts=2024-10-18T01:07:18.186139Z caller=ruler.go:727 level=error msg="unable to list rules" err="unable to read dir /tmp/cortex/rules: open /tmp/cortex/rules: no such file or directory"
```
But, it doesn't abort.

Reply:
OK thanks.